bloom-housing/bloom
# feat: refactor listing actions file

# Pull Request Template

## Issue Overview

This PR addresses [#307](https://github.com/metrotranscom/doorway/issues/307)


## Description

This PR has no functional changes, but I had to refactor the existing listings action panel to support the toggling we need and decided to put that in a separate PR to keep things small.

## How Can This Be Tested/Reviewed?

Try as I might I struggled to get the git history on the `Aside` file to stay.

I renamed `Aside` to `ListingFormActions`, separated the logic bit about which buttons should be shown when from the buttons themselves, and consolidated some duplicate code. Comparing the two files will show they are very similar.

Also fixed a bug where the status was not expanding the full width.

No functionality should change.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
